raster-foundry/raster-foundry
# Remove RasterSource cache, construct better cache keys

## Overview

This PR probably fixes a cache collision that was happening when project layers have identical AOIs. I say "probably" because I've so far reproduced the bug a single time in four hours, and I was unable to follow the same steps to reproduce it after that.

### Checklist


## Testing Instructions

- go clone the embed widget https://github.com/raster-foundry/embed-widget/
- edit `webpack.dev.js` in the widget to use something other than `:8080`
- edit `index.js` to use your local tile server and api server (`:8081` and `:9100`)
- create a project
- create a layer in addition to the default layer on that project
- create  a scene in two different layers by importing a COG of the Vermont tiff (don't use the same scene -- make two different ones)
- set one of the layers' visualizations to single band
- leave the other one as is or do whatever you want with it, just make sure it's not the same single band as the other one
- publish your project with a map token
- navigate to: `localhost:<embed port>/?projectID=<your project id>&mapToken=<your map token>&type=comparison&layers=true&analyses=`
- pick one layer for each side
- you shouldn't get any weird visualization artifacts where one side has the other side's tiles showing up

Closes #5000 again
